packages/material-ui/src/Popover/Popover.test.js

```diff
@@ -1,22 +1,56 @@
 import React from 'react';
 import { assert } from 'chai';
 import { spy, stub, useFakeTimers } from 'sinon';
-import PropTypes from 'prop-types';
 import {
-  createShallow,
   createMount,
   describeConformance,
   findOutermostIntrinsic,
   getClasses,
 } from '@material-ui/core/test-utils';
-import consoleErrorMock from 'test/utils/consoleErrorMock';
 import Grow from '../Grow';
 import Modal from '../Modal';
 import Paper from '../Paper';
-import Popover from './Popover';
+import Popover, { getOffsetLeft, getOffsetTop } from './Popover';
+import { useForkRef } from '@material-ui/core/utils/reactHelpers';
+
+const mockedAnchorEl = () => {
+  const div = document.createElement('div');
+
+  stub(div, 'getBoundingClientRect').callsFake(() => ({
+    width: 100,
+    height: 58,
+    left: 160,
+    top: 160,
+    bottom: 218,
+    right: 260,
+  }));
+  return div;
+};
+
+const FakePaper = React.forwardRef(function FakeWidthPaper(props, ref) {
+  const paperRef = React.useRef(null);
+  const handleRef = useForkRef(ref, paperRef);
+
+  React.useEffect(() => {
```

Comment:
I would recommend doing this with a callback ref instead. Otherwise this test relies on `act` which shouldn't be required in a "natural" environment.